source/s3_client.c

```diff
@@ -319,9 +322,22 @@ PyObject *aws_py_s3_client_new(PyObject *self, PyObject *args) {
         signing_config = &default_signing_config;
     }
 
+    struct aws_byte_cursor *network_interface_names = NULL;
+    int num_network_interface_names = 0;
+
+    if (network_interface_names_py != Py_None) {
+        if (!PyList_Check(network_interface_names_py)) {
+            PyErr_SetString(PyExc_TypeError, "Expected network_interface_names to be a list");
+            return NULL;
+        }
+        Py_ssize_t listSize = PyList_Size(network_interface_names_py);
```

Comment:
Use the [Py_Sequence API](https://docs.python.org/3/c-api/sequence.html) so you can accept tuples too

Reply:
Thanks, updated to sequence.